### DISCUSSION TITLE
Frame limiter not working when running the game in single threaded mode

### DISCUSSION THREAD

**Discussion by Sebastian-Zamora** [ppy/osu#27528] [Q&A]
### Type

Game behaviour

### Bug description

I've been trying to limit the frame rate to x2 or x4 while running the game in single threaded and only Vsync seems to work, only if I switch back to multithreaded the framerate actually gets capped. Tried this in 2 different computers and got the same results.

The main reason for me to run the game in single threaded mode is that it heavily reduces the cursor latency on my computer and makes it comparable to stable. While running the game with no limiter provides a very smooth experience, I occasionally encounter micro freezes that can be annoying, which do not occur in multi-threaded mode. However, the cursor latency is considerably worse for me in that mode.

### Version

2024-130-2-lazer

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14128322/compressed-logs.zip)

**Comment by theyareonit** [comment]
Experiencing the same issue. Setting the framerate limiter to refresh rate x2 on single-threaded mode results in a 960fps cap on 240hz, and a 480fps cap on 120hz. 
[compressed-logs.zip](https://github.com/ppy/osu/files/14532598/compressed-logs.zip)

**Comment by peppy** [comment]
This means that you have vsync forced externally.

**Reply by theyareonit** [reply]
I don't think this is what's happening. Wouldn't the framerate always be stuck at your refresh rate if it was vsync? What me and the OP are experiencing is that, only when single-threaded mode is enabled, the framerate limit is much higher than it should be. In my case, my framerate is doubled on single-threaded mode compared to multi-threaded mode.
